listener: reject duplicated Ipv4-mapped address and Ipv4 addresss (#20504)
envoyproxy/envoy: docs/root/version_history/current.rst

* http: lazy disable downstream connection reading in the HTTP/1 codec to reduce unnecessary system calls. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.http1_lazy_read_disable`` to false.
* http: now the max concurrent streams of http2 connection can not only be adjusted down according to the SETTINGS frame but also can be adjusted up, of course, it can not exceed the configured upper bounds. This fix is guarded by ``envoy.reloadable_features.http2_allow_capacity_increase_by_settings``.
* http: when writing custom filters, `injectEncodedDataToFilterChain` and `injectDecodedDataToFilterChain` now trigger sending of headers if they were not yet sent due to `StopIteration`. Previously, calling one of the inject functions in that state would trigger an assertion. See issue #19891 for more details.

> * listener: reject to listen both on the ipv4-mapped ipv6 address and the corresponding ipv4 address. The runtime flag `envoy.reloadable_features.reject_duplicated_ipv4_mapped_address_and_ipv4_address` is added for this.

Reviewer: please use double backtics for literals

i thought we add some ci to prevent this for version history - but it doesnt seem to be working - ill raise a PR to clean up the others in this file

Author: got it, thanks!

> * listener: reject to listen both on the ipv4-mapped ipv6 address and the corresponding ipv4 address. The runtime flag ``envoy.reloadable_features.reject_duplicated_ipv4_mapped_address_and_ipv4_address`` is added for this.

Reviewer 2: ```suggestion
* listener: reject to listen both on the IPv4-mapped IPv6 address and the corresponding IPv4 address. The runtime flag ``envoy.reloadable_features.reject_duplicated_ipv4_mapped_address_and_ipv4_address`` can be used to revert this behavior.
```

Author: done, thanks
